src/OpenTelemetry/Trace/Builder/TracerProviderBuilderExtensions.cs

```diff
@@ -24,6 +24,9 @@ public static class TracerProviderBuilderExtensions
     /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
     /// <param name="enabled">Enabled or not. Default value is <c>true</c>.</param>
     /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
+#if NET7_0_OR_GREATER
+    [RequiresDynamicCode("SetErrorStatusOnException is neither supported in MonoRuntime nor in nativeAOT environment.")]
```

Comment:
Why add this annotation when you have already updated `ExceptionProcessor` ctor to throw an exception at build time for non-supported scenarios?

Reply:
@utpilla The exception will be runtime not build time. This warning will be build time. So I think that may be reason enough to have it. But I don't feel passionately about it if we want to just go with the runtime logic I'm OK with that. Maybe having this attribute would be a good reminder to clean it up in the future if .NET gives an AOT-safe way to do it?

Reply:
> The exception will be runtime not build time.

Technically, it is runtime, but it does let us achieve what we want (which is not to crash the application once it's up and running). Since the SDK would throw at setup/application startup, it would essentially be the equivalent of user's app never being started.

There is another issue with the attribute which it's not relevant to the actual issue here. There is no involvement of dynamic code here if I understand [this](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5374#discussion_r1501287922) correctly. Why resort to using misleading attributes when we can achieve what we want using ctor approach?

Reply:
> 
> Since the SDK would throw at setup/application startup, it would essentially be the equivalent of user's app never being started.
> 
Thank you! I removed the AOT attribute because the above mentioning rationale.